<?xml version="1.0"?>
<launch>
  <arg name="world_name" default="basic_R3"/>
  <arg name="enable_logging" default="false"/>
  <!--arg name="image_def" default="vi_sensor/camera_depth/depth"/-->

 <!--node pkg="geometric_controller" name="publish_goal_to_chase" type="publish_goal_to_chase.py" respawn="true" output="screen"/-->
 <node pkg="geometric_controller" name="publish_targets_using_qcp" type="sensor_placement_using_qcp_ros.py" respawn="true" output="screen"/>
 <node pkg="geometric_controller" name="move_turtlebot" type="move_turtlebot.py" respawn="true" output="screen"/>
 <!--node pkg="geometric_controller" name="publish_targets" type="sensor_placement_nlopt_R3_ROS.py" respawn="true" output="screen"/-->

  <group ns="firefly1">
      <param name="uav_name" value="firefly"/>
      <param name="number" value="1"/>
      <param name="uav_velocity" value="3.0"/>
      <param name="publish_frequency" value="100.0"/>
      <param name="initial_hovering_height" value="3.0"/>
      <node pkg="geometric_controller" name="PositionVelocityController" type="position_velocity_controller_multiquadlaunch.py" respawn="true"/>
      <node pkg="geometric_controller" name="navfunc3d_nobstacle" type="3DNF_MRSpaper_ros_R1.py" respawn="true" output="screen"/>
      <!--node pkg="geometric_controller" name="navfunc3d_nobstacle" type="3DNF_MRSpaper_ros.py" respawn="true" output="screen"/-->
      <remap from="odometry" to="odometry_sensor1/odometry" />	     
  </group>

  <group ns="firefly2">
      <param name="uav_name" value="firefly"/>
      <param name="number" value="2"/>
      <param name="uav_velocity" value="3.0"/>
      <param name="publish_frequency" value="100.0"/>
      <param name="initial_hovering_height" value="3.0"/>
      <node pkg="geometric_controller" name="PositionVelocityController" type="position_velocity_controller_multiquadlaunch.py" respawn="true"/>
      <node pkg="geometric_controller" name="navfunc3d_nobstacle" type="3DNF_MRSpaper_ros_R1.py" respawn="true" output="screen"/>
      <!--node pkg="geometric_controller" name="navfunc3d_nobstacle" type="3DNF_MRSpaper_ros.py" respawn="true" output="screen"/-->
      <remap from="odometry" to="odometry_sensor1/odometry" />	     
  </group>

  <group ns="firefly3">
      <param name="uav_name" value="firefly"/>
      <param name="number" value="3"/>
      <param name="uav_velocity" value="3.0"/>
      <param name="publish_frequency" value="100.0"/>
      <param name="initial_hovering_height" value="3.0"/>
      <node pkg="geometric_controller" name="PositionVelocityController" type="position_velocity_controller_multiquadlaunch.py" respawn="true"/>
      <node pkg="geometric_controller" name="navfunc3d_nobstacle" type="3DNF_MRSpaper_ros_R1.py" respawn="true" output="screen"/>
      <!--node pkg="geometric_controller" name="navfunc3d_nobstacle" type="3DNF_MRSpaper_ros.py" respawn="true" output="screen"/-->
      <remap from="odometry" to="odometry_sensor1/odometry" />	     
  </group>

</launch>
